tests/ResultQueryTest.php

```diff
@@ -588,4 +588,171 @@ public function testQueryStreamFromLazyConnectionWillErrorWhenConnectionIsClosed
 
         $connection->close();
     }
+
+    protected function checkMaxAllowedPacket($connection, $min = 0x1100000)
+    {
+        return $connection->query('SHOW VARIABLES LIKE \'max_allowed_packet\'')->then(
+            function ($res) use ($min, $connection) {
+                $current = $res->resultRows[0]['Value'];
+                if ($current < $min) {
+                    return $connection->query('SET GLOBAL max_allowed_packet = ?', [$min])->otherwise(
+                        function (\Throwable $e) use ($min, $current) {
+                            throw new \Exception(
+                                'Cannot set max_allowed_packet to: ' . $min . ' ' .
+                                'current: ' . $current . ' error: ' . $e->getMessage()
+                            );
+                        }
+                    );
+                }
+                return \React\Promise\resolve();
+            }
+        )->then(
+            function () {
+                return true;
+            }
+        );
+    }
```

Comment:
You said in one of your last comments:

>  I tried, but user does not have permission to change max_allowed_packet

In here you're trying to overwrite `max_allowed_packet`, so I'm a little confused if it works or not. Or maybe this is a leftover? It seems like your tests will never overwrite  `max_allowed_packet` in this case because `$min` never surpasses `$current`, right?

Reply:
Yes, it will try. Maybe useful for someone with smaller `max_allowed_packet` but permission to change it. If it does not succeed it will mark test as incomplete.

Reply:
Yeah, I agree, we should probably not set GLOBAL configuration. I'll rewrite tests to simply check if `max_allowed_packet` is set to large enough value for tests, otherwise tests will be marked as incomplete.
I will update the `Readme.md` file to include `max_allowed_packet` setting.